src/txadmission.cpp

```diff
@@ -698,17 +776,49 @@ bool ParallelAcceptToMemoryPool(Snapshot &ss,
 
                     if (!view.HaveCoin(txin.prevout))
                     {
+                        // Are inputs missing because we already have the ptx?
+                        for (size_t out = 0; out < tx->vout.size(); out++)
+                        {
+                            // Optimistically just do efficient check of cache for
+                            // outputs.
+                            if (pcoinsTip->HaveCoinInCache(COutPoint(tx->GetHash(), out)))
+                            {
+                                // return state.Invalid(false, REJECT_DUPLICATE,
+                                //                     "txn-already-known");
+                                // debugger->AddInvalidReason("txn-already-known");
+                                txnAlreadyKnown = true;
+                                break;
+                            }
+                        }
+                        if (debugger)
+                        {
+                            debugger->mineable = false;
+                            debugger->futureMineable = false;
+                            debugger->AddInvalidReason("input-does-not-exist: " + txin.prevout.hash.ToString() + ":" +
+                                                       std::to_string(txin.prevout.n));
+                        }
                         // fMissingInputs and not state.IsInvalid() is used to detect this condition, don't set
                         // state.Invalid()
                         *pfMissingInputs = true;
                         break; // There is no point checking any more once one fails, for orphans we will recheck
```

Comment:
In debugger mode, do we want to check all the inputs, following your philosophy of finding all the problems with the tx?

Reply:
yes, this is break should not happen in debug mode, good catch